### PR TITLE
Rework UX guidelines to be in distinct folders and files

### PR DESCRIPTION
**Pull request by JacklynBiggin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

To prepare for the launch of the Woo Developer Docs site, this PR reworks the user experience guidelines section. This is due to us expecting additional categories of guidelines to be released. This PR changes:

* Payments guidelines are split into individual files contained within `docs/ux-guidelines-payments`.
* Theme guidelines are split into individual files contained within `docs/ux-guidelines/themes`. The original blurb from that page has been moved into the category description as it is broadly applicable to all content.
* The `docs/user-experience` folder is renamed to `docs/ux-guidelines-extensions` for consistency. Additionally, the title of the category is renamed to `Extension Guidelines` to better reflect what they contain.
* The titles of pages within the `docs/ux-guidelines-extensions` pages have been updated to better reflect their purpose and new category name. For example, `User Experience Guidelines - Notices` is now called `WooCommerce Extension Guidelines - Notices`.
* The contents of `user-experience-guidelines.md` has been moved to the folder's `README.md` file, so they'll appear on the category page itself. The links themselves have been removed as they'll be automatically shown under the descriptive text.

When implemented into the docs site, they will appear as shown below.

CleanShot 2024-01-11 at 22 23 49@2x

**NOTE: prior to merging this PR, the docs manifest will need to be regenerated. This will happen once we're happy with the new folder structure.**

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check all modified markdown files are consistent and have a sensible folder structure.
2. Ensure no typos or grammar mistakes
3.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by JacklynBiggin** [comment]
Updated to also move the product editor guidelines into their own folder (split into separate files for consistency).

File: docs/ux-guidelines-payments/payment-button-size.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- Large: Height 56
+
+### Anatomy
+
+Payment buttons consist of up to three elements: a button component, payment logo, and optional label.
+[Anatomy.png]
+
+Don’t place any text outside the surface area of the button.
+
+![Cart layout 2](https://woo-docs-multi-com.go-vip.net/docs/wp-content/uploads/sites/3/2024/01/Cart-layout-2.png")
```
Comment (elizaan36) [inline docs/ux-guidelines-payments/payment-button-size.md]: @JacklynBiggin For some reason this image isn't loading. Let me know if you need me to resend.

File: docs/ux-guidelines-themes/theme-customization.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+post_title: WooCommerce Theme Guidelines - Customization
+menu_title: Customization
+---
+
+Themes have to rely on the customizer for any type of initial set up. Specific onboarding flows are not permitted.
```
Comment (elizaan36) [inline docs/ux-guidelines-themes/theme-customization.md:8]: Hm this only applies to non-block themes. Would it be possible to add something like the following in helper text (or notice) at the top?

> This page of the guidelines applies to development of non-block themes only. For more specific guidance on development of block themes, refer to the [WordPress Developer's Guide to Block Themes](https://learn.wordpress.org/course/a-developers-guide-to-block-themes-part-1/).

**Review by elizaan36 — APPROVED** [review]
Left a couple comments but otherwise looks good to me. I had a look at the product editor guidelines as well but it might be worth having @pmcpinto or @jarekmorawski's review as well since I'm not as close to the development of those pages.

File: docs/ux-guidelines-payments/payment-button-size.md (hunk 1 above)
Comment (JacklynBiggin) [reply docs/ux-guidelines-payments/payment-button-size.md]: Good catch! Removed the " - should be fixed!

File: docs/ux-guidelines-themes/theme-customization.md (hunk 2 above)
Comment (JacklynBiggin) [reply docs/ux-guidelines-themes/theme-customization.md:8]: done!

**Review by brentmackinnon — APPROVED** [review]

**Comment by pmcpinto** [comment]
cc @louwie17 in case you or the rest of the team want to check the changes.

**Pull request merged**